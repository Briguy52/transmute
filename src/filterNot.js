import curry from "./curry";

/**
 * Remove values for which `predicate` returns `true`.
 *
 * @example
 * // returns List [ 1, 3 ]
 * filterNot(
 *   (n) => n % 2 === 0,
 *   List.of(1, 2, 3)
 * );
 *
 * @param {Function} predicate returns `true` if a value should be excluded.
 * @param {Iterable} subject to filter.
 * @return {Iterable} without values that matched `predicate`.
 */
function filterNot(predicate, subject) {
  return subject.filterNot(predicate);
}

export default curry(filterNot);
