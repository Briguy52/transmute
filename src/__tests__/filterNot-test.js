import filterNot from "../filterNot";
import { List, Map } from "immutable";

describe("transmute/filterNot", () => {
  const takeOdd = filterNot(n => n % 2 === 0);

  it("filterNots Lists", () => {
    expect(takeOdd(List.of(1, 2, 3))).toMatchSnapshot();
  });

  it("filterNots Maps", () => {
    expect(takeOdd(Map.of("one", 1, "two", 2, "three", 3))).toMatchSnapshot();
  });
});
