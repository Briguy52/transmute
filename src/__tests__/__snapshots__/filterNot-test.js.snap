// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transmute/filterNot filterNots Lists 1`] = `
Immutable.List [
  1,
  3,
]
`;

exports[`transmute/filterNot filterNots Maps 1`] = `
Immutable.Map {
  one: 1,
  three: 3,
}
`;
